sdks/python/apache_beam/io/mongodbio_test.py

```diff
@@ -0,0 +1,187 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import logging
+import unittest
+
+import mock
+
+import apache_beam as beam
+from apache_beam.io import ReadFromMongoDB
+from apache_beam.io import WriteToMongoDB
+from apache_beam.io import source_test_utils
+from apache_beam.io.mongodbio import _BoundedMongoSource
+from apache_beam.io.mongodbio import _GenerateObjectIdFn
+from apache_beam.io.mongodbio import _MongoSink
+from apache_beam.io.mongodbio import _WriteMongoFn
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+
+
+class MongoSourceTest(unittest.TestCase):
+  @mock.patch('apache_beam.io.mongodbio._BoundedMongoSource'
+              '._get_document_count')
+  @mock.patch('apache_beam.io.mongodbio._BoundedMongoSource'
+              '._get_avg_document_size')
+  def setUp(self, mock_size, mock_count):
+    mock_size.return_value = 10
+    mock_count.return_value = 5
+    self.mongo_source = _BoundedMongoSource('mongodb://test', 'testdb',
+                                            'testcoll')
+
+  def test_estimate_size(self):
+    self.assertEqual(self.mongo_source.estimate_size(), 50)
+
+  def test_split(self):
```

Comment:
You should be able to use assert_sources_equal_reference_source() here.
https://github.com/apache/beam/blob/master/sdks/python/apache_beam/io/source_test_utils.py#L116

Reply:
done